data/core/units/elves/Champion.cfg

```diff
@@ -24,8 +24,9 @@
         icon=attacks/greatsword-elven.png
         type=blade
         range=melee
-        damage=9
+        damage=8
         number=5
+        accuracy=10
```

Comment:
Why modify accuracy directly in the attack like this, rather than adding a weapon special?

Reply:
What is the purpose of this stat existing then? It exists so im using it. Not gonna do weapon specials for stuff that exists already as a stat.

Reply:
well ok, does it show up anywhere in-game? How is the player supposed to know about it being there?

Reply:
Why is this question for me? If players dont see unit stats then thats UI probelm. 

It is visible, not very well explained, but visible.

Reply:
There are no unit tests for `accuracy`, it's not used directly in mainline WML (there's some use of `increase_accuracy`, but not `accuracy`), and it's used exactly once in the entirety of Ageless Era. It has the C++ code to support it in the combat calculations and store_unit/unstore_unit, but that's about it.

If we're going to support it properly, it should have tests and UI, and there are already two issues about that (#2517 and #5860). However, as we're getting close to the release candidates for 1.18, it seems late in the development cycle to add those to the roadmap.

Using a weapon special instead would be a simple way to ensure it works well with the current UI, and at this point in the 1.18 cycle that seems the best choice to me.

Reply:
>  If players dont see unit stats then thats UI problem.

I agree with Hejnewar here. The UI should be enhanced to augment the UX of the game and I dont see this is a argumentative blocker for the balance patch. Also, the accuracy of +10% is shown in the sidebar under the unit's attack as a `+10%/0%` which means +10% to CTH on attack, and 0% to enemy CTH on defense.

Reply:
So, because no one uses it, it was never a feature that was made stable and its usage is discouraged because it is unstable? Since I wanted to use it in many places, there was an issue for few years. It wasnt made any less bad. Now I want to use it. 

And well I will use it anyway so maybe that issue is finally resolved.

Reply:
If we were at 1.17.5 or 1.19.5, then I'd agree that using it is the best way to get the issues tested and resolved.

But we're about 3 months before string-freeze with other UI issues that need fixing, and still working on getting the event system and ability filter changes tested. I hope that the accuracy feature is stable, but if it isn't, then I doubt it will be fixed for the 1.18 branch.

Reply:
What's the problem with the current UI for it?

Also, "you can't use features that have no unit tests" isn't a requirement that I'm aware of? It would certainly be good for a unit test to be added for it, but as long as Hejnewar has tested it himself and can confirm that it does work the way it's supposed to, then IMO that should be good enough.

Reply:
It just doesnt say what it does, if we threat it like any other stat that doesnt need explaining thats fine but right now acuraccy x / parry y is all it says. We can assume its fine but usually there is some confusion around it.

Reply:
Changing the tooltip in #5860 to say `<b>Accuracy:</b> +10% bonus to chance to hit` would solve a lot of my concerns about the UI. Not difficult, but without @cooljeanius flagging this issue it wouldn't have happened before the string-freeze. I'm still not sure that it will happen in time for the string freeze.

> Also, "you can't use features that have no unit tests" isn't a requirement that I'm aware of?

It's more a general feel of how many of the edge cases have been tested, how it interacts with other abilities that change the chance-to-hit stats, for SP how it interacts with the PRNG options. The existence of unit tests is reassuring, the amount that it gets used in Mainline + Ageless is my main measure for guessing how many of the edge cases are known.

For PRs, my sad lesson from experience is to assume that testing was minimal unless someone say otherwise.

Reply:
@Hejnewar have you tested using the Champion with the `accuracy` attribute added? Or @ProditorMagnus have there been any issues with it in Ageless Era?

Reply:
Rashy Era uses it out of actively developed source eras. With 4.30 it was also added to several other units for balancing. I would encourage tooltip change too, but values are visible from attack dialog so that should not be a blocker. @andrzejskutela uses it for boss units too.

Reply:
I was looking at Ageless 4.28. I confirm, I'm a lot happier about this PR after looking at the current Ageless.

Reply:
Getting it to core unit should also help to speed up getting it to wmlunits.

Reply:
> Getting it to core unit should also help to speed up getting it to wmlunits.

Is there a separate issue for that?

Reply:
for the record, since my name was mentioned
[accuracy + parry visual](https://ibb.co/QXYJR6y)
[hover](https://ibb.co/x5Tw6NV)

Reply:
Shouldnt be.

Reply:
Well for now I will merge it, if there is still some discussion needed it can be moved to an issue and if something is really wrong I can always change the change.

Reply:
for cross-referencing purposes, I opened #7895 for the `wmlunits` part